pkg/arvo/sys/zuse.hoon

```diff
@@ -5296,6 +5296,23 @@
     ?~  a  b
     ::
     $(a r.a, b [n.a $(a l.a)])
+  ::  +run: apply gate to transform all values in place
+  ::  without modifying their type
+  ::
+  ++  run
+    |=  [a=(tree item) b=$-(val val)]
+    |-  ^+  a
+    ?~  a  a
+    [n=[key.n.a (b val.n.a)] l=$(a l.a) r=$(a r.a)]
+  ::  +all: apply logical AND boolean test on all values
+  ::
+  ++  all
+    |=  [a=(tree item) b=$-(item ?)]
+    ^-  ?
+    |-
+    ?~  a
+      &
+    ?&((b n.a) $(a l.a) $(a r.a))
```

Comment:
Would be pretty easy to make this short-circuit

Reply:
What do you mean? It will end early if any check returns false.

Reply:
It'll not look at its children, but if the top node is true, then even if the left branch is false, the right branch still gets run.

Reply:
wait no i'm wrong aren't it

Reply:
Yeah no this is correct

Reply:
Interesting, okay. Worth noting that I took this directly from +in in hoon.hoon, so that should short-circuit earlier too. Likely also the one in +by

Reply:
```
> &(& ~&(%foo |))
%foo
%.n
> 
> &(| ~&(%foo |))
%.n
```

Reply:
If it was not short-circuiting, that second example would also slog.

Reply:
> Yeah no this is correct
By this I meant your code is correctly short-circuitying